### ISSUE TITLE
could not update due to long inactivity (xbps < 0.54)

### ISSUE THREAD

**Issue by erdosnew** [void-linux/xbps#251]
hi, i've a system that i haven't booted for a while (since May 2019).
NOw i couldn't update the pkg or the system due to the fact that my 'xbps is ver xbps-0.53_9.

sh-5.0# xbps-install -Su
[*] Updating `https://mirror.clarkson.edu/voidlinux/current//i686-repodata' ...
i686-repodata: 1652KB [avg rate: 12MB/s]
[*] Updating `https://alpha.de.repo.voidlinux.org/current/i686-repodata' ...
i686-repodata: 1652KB [avg rate: 1843KB/s]
..
then nothing happened, how do i proceed with updating?

**Comment by xtraeme** [comment]
Use the static xbps

**Comment by Vaelatern** [comment]
Someone just updated a system from 4 years ago, they used the static xbps to update their system.

**Comment by erdosnew** [comment]
i downloaded 'xbps-static-static-0.56_5.x86_64-musl.tar.xz' from the site and extracted the files into /usr/bin and /var/db folders.

while rerun the xbps, i got the following error:  

sh-5.0# xbps-install -Su
sh: /bin/xbps-install: cannot execute binary file: Exec format error
sh-5.0# /usr/bin/xbps-install -Su
sh: /usr/bin/xbps-install: cannot execute binary file: Exec format error

is there something i missed?

**Comment by erdosnew** [comment]
btw, are the package for 'musl'?  I run Void on a 32bit machine with standard C

**Comment by Vaelatern** [comment]
So you are running on a 32 bit machine and downloaded the 64 bit static xbps, I'd recommend installing the i686 version instead.

Just set XBPS_ARCH=i686 and all should work fine.

**Comment by erdosnew** [comment]
may i ask how to 'set XBPS_ARCH=i686'?

**Comment by Vaelatern** [comment]
Try `XBPS_ARCH=i686 /usr/bin/xbps-install.static -Su` ?

**Comment by the-maldridge** [comment]
Seems pretty reasonable.  I updated a system last booted in 2016 using this method.  I did not unpack the binaries onto the rootfs, instead preferring to keep them in a subdirectory.

I also prefer to export variables and then use `sudo -E`, but that is personal preference as well.  It should be pretty straightforwards, but I did chose to pre-emptively remove python2 and a number of packages that I know no longer exist.
